staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -172,8 +192,10 @@ func (t *envelopeTransformer) TransformFromStorage(ctx context.Context, data []b
 			Annotations: encryptedObject.Annotations,
 		})
 		if err != nil {
+			span.AddEvent("DEK decryption failed", attribute.String("error", err.Error()))
```

Comment:
consider using span.RecordErr as well, and removing the error attribute.

Applies here and below.

Reply:
@dashpole I'm using `k8s.io/component-base/tracing` for trace here and `*tracing.Span` doesn't have a `RecordError` method.

Reply:
Can you update componen-base/tracing to make it possible?  Does this work (in component-base)?:

```diff
type Span struct {
+	trace.Span
-	otelSpan trace.Span
	utilSpan *utiltrace.Trace
}
```

Otherwise, you can just add RecordError to delegate to the OpenTelemetry span.

Reply:
I've added `RecordError` to delegate to otel span. LMK what you think!

Reply:
Component-base changes LGTM